session_test.go

```diff
@@ -981,13 +991,7 @@ func TestSession_PingOfDeath(t *testing.T) {
 }
 
 func TestSession_ConnectionWriteTimeout(t *testing.T) {
-	// Disable keepalives so they don't detect the failed connection
```

Comment:
Good call in making sure these fail for the right reasons. Would you mind adding a test case that exercises your new timeout?